test/testdata/desugar/block_given.rb

```diff
@@ -52,3 +52,13 @@ def method_with_implicit_block_param()
   SomethingElse.block_given?
   block_given?("with parameter")
 end
+
+# Outside of a method, none of these shuold desugar to special calls to `block_given?`
```

Comment:
This case was already partly covered in `test/testdata/infer/block_given.rb`, but I wanted a more thorough desugar-specific test.

The test passes both before and after the implementation change.